_studio/mfx_lib/encode_hw/h265/src/mfx_h265_encode_vaapi.cpp

```diff
@@ -1786,16 +1786,17 @@ mfxStatus VAAPIEncoder::QueryStatus(Task & task)
     {
         VASurfaceStatus surfSts = VASurfaceSkipped;
 
-        vaSts = vaQuerySurfaceStatus(m_vaDisplay, waitSurface, &surfSts);
-
-        MFX_CHECK_WITH_ASSERT(VA_STATUS_SUCCESS == vaSts, MFX_ERR_DEVICE_FAILED);
+        m_feedbackCache.erase(m_feedbackCache.begin() + indxSurf);
+        guard.Unlock();
 
-        if (VASurfaceReady == surfSts)
         {
-            m_feedbackCache.erase(m_feedbackCache.begin() + indxSurf);
-            guard.Unlock();
+            MFX_AUTO_LTRACE(MFX_TRACE_LEVEL_EXTCALL, "vaSyncSurface");
+            vaSts = vaSyncSurface(m_vaDisplay, waitSurface);
```

Comment:
We should not use vaSyncSurface with input surface for encoder synchronization, but vaMapBuffer with bitstream

Reply:
I partially disagree. As of now this is a designated uAPI to sync tasks. Yes, we move to use vaMapBuffer instead for the encoders, but this is not official way yet. Thus, I would like to have a commit which uses vaSyncSurface in a history. Current change is definitely better than vaQuerySurfaceStatus (how we managed to have that by the way?!). I suggest to merge and then switch to vaMapBuffer.

Reply:
Agree